### ISSUE TITLE
`join_return_with_assignment` tests should include assignment patterns

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#4157]
Child issue of https://github.com/dart-lang/linter/issues/3624

I think there should be test cases like:

```dart
var (a, b) = record; // variable declaration
return a;

// also
(a, b) = record; // just an assignment
return a;
```

in which the lint rule would _not_ report that the return should be joined with the assignment.

(There are also no test cases with multiple assignment expressions in one statement, like

* `a = b = 7; return b;` or
* `var a = 8, b = 9;  return b;`

)

**Comment by scheglov** [comment]
We don't report this lint even for a single declared local variable.
```dart
int f(int a) {
  var a = 0;
  return a;
}
```
We could, I guess, although this is not an _assignment_.

**Comment by scheglov** [comment]
As for multiple assignments, e.g.
```dart
int f1(int a, int b) {
  a = b = 0;
  return a;
}

int f2(int a, int b) {
  a = b = 0;
  return b;
}
```

Semantically it is not wrong to inline as `return a = b = 0;` in both cases.
But it definitely feels weird.
And we report the lint only for one of them, `f1`.

**Comment by srawlins** [comment]
Ah good analysis; we don't need to rock the boat for patterns; if _declarations_ aren't covered, then they aren't covered. No biggie. But the assignment pattern (identifier pattern?) I think should still have a test.

**Comment by scheglov** [comment]
Tests were added.
